### DISCUSSION TITLE
RPI 2 B v1.1-Experiments

### DISCUSSION THREAD

**Discussion by digitalelements** [JarvyJ/pislide-os#19] [Show Off Your Project!]
Hi There,

This is more of a report of observations rather than showing off a PiSlide project.

I happen to have a Pi 2 B v1.1 to experiment with. The first boot wasn’t quit successful. but a reboot
seemed to put things back on track. The restart seemed to be “required” as I did the imaging and first run of PiSlide
8-10 times today so far. When I got to the steps of adding my own images, I experienced what I thought was a hang or a crash of the Pi. It was taking forever to start the Slideshow ( so I thought). What I finally realized was the time needed to cache/prepare the slideshow at startup before it was visibly running. I’ve narrowed it down to the numbers in the ini file. I have 12 slides with the default settings for duration and transition which was roughly 2 min 40 sec. Things are running great now that I’ve got my head around it better. I do wonder about the repeat time of the entire collection ? The slide show takes the same amount of time to start again (2:40 at rest) before the show starts again. will there be a way to shorten the downtime ? … Really love the PiSlide concept and, I look forward to using it more and more. I’m also sure this is likely very ambitious for the Pi 2 so I look forward to trying things on much better hardware too.

**Comment by JarvyJ** [comment]
I'm glad you've tried it out! Yeah to get it to load faster initially, you can resize the images to fit the screen. That way the Pi doesn't spend cycles resizing the images the first time around.

Also, the image viewer actually caches images when trying to load the image for the first time, not all of them on boot. But, I think there's a bug with the timer code where the timer starts while the image is still being downsized but not when it starts being displayed. I'll take a look at that.

What might be happening on Pi 2 hardware is it take so long to downsize, it actually ends up resizing all of them before it can even display them.

I might also try and get the image viewer to _show something_ while downsizing so it's not just a blank screen on older Pis.

Thanks for letting me know about your experience! It will help make things better in the future!

**Reply by digitalelements** [reply]
Ahhh .. good info. I'll certainly experiment with optimizing the images for the screen. Are the images still cached if they are the optimal size for the screen ? .. or only when resizing is required ?

**Reply by JarvyJ** [reply]
They should only get cached when resizing is required.

**Comment by digitalelements** [comment]
Hey Jarvy,  spent a good amount of time experimenting with things over the weekend . I built slideshows with a number of different files sizes and formats. All ran very nicely. I spied on the cache to see what screen size was established to see what PiSlide wanted to work with. Just had to make sure to give the Pi time to prepare everything at the start. I was working with 12 slides and the default slide duration and transitions. So, about 2:40 runtime. My only hope is to have the slideshows run more continuously in the future. With the 12 sides and default duration and transition values, all slide shows run for 2:40 and then dark screen for 2:40 before the next showing.... Thanks !

**Comment by freddieFishCake** [comment]
One possible strategy to minimise the expensive re-generation of optimal size images for the cache at every reboot might be to only re-generate images if either the screen size has changed (delete entire cache and re-generate all images), or if there is no corresponding image in the cache for a given original (new image) or if the timestamp of the original is greater than the optimised image (updated original).
Thus on most reboots no image optimisation would be needed.

**Reply by JarvyJ** [reply]
I really like the screen size/timestamp for cache invalidation. It covers all the use cases nicely. I'll be sure to work it in for the next release! Thanks for the suggestion!

**Reply by freddieFishCake** [reply]
I should have said screen size or rotation changed!
